Pilha.java

```diff
@@ -0,0 +1,74 @@
+public class Pilha {
+
+    public Object[] pilha; 
+    public int posicaoPilha;
+    public int tamMax;
+
+    public Pilha() {
+        this.posicaoPilha = -1;
+        this.pilha = new Object[10];
+        tamMax = 10;
+    }
+
+    public boolean pilhaVazia() {
+        //isEmpty
+        if (this.posicaoPilha == -1) {
+            return true;
+        }
+        return false;
+    }
+    
+    public boolean pilhaCheia() {
+    	//isFull
+    	if (this.posicaoPilha == tamMax){ 
+    		return true;
+    	} else {
+    		return false;
+    	}
+    }
+    
+
+    public int tamanho() {
+        //is
+        if (this.pilhaVazia()) {
+            return 0;
+        }
+        return this.posicaoPilha; 
```

Comment:
Essa linha apresenta um problema, como a posiçãoPilha começa com -1, logo quando tiver apenas um elemento na pilha ela vai retornar o valor de 0. Se tiver 2 elementos na pilha ele vai retornar o valor de 1. Para corrigir isso, o retorno deve ser: return this.posicaoPilha+1;